### PR TITLE
SSH for Postgres Destination

### PR DESCRIPTION
**Pull request by cgardens**
## What
* Follows on work from SSH Postgres Source https://github.com/airbytehq/airbyte/pull/5742
* @Phlair as far as I can tell, all tests are passing except those that require normalization. Can you rebase onto this branch so that we can get to green and release this.

## How
* Slightly different approach than I took on the Postgres Source. I use a decorate to wrap the destination methods in SSH tunnels. I think I like this pattern better. Curious to hear opinions.

## Recommended reading order
1. `SshDestination.java`
2. `SshWrappedJdbcDestination.java`

## Pre-merge Checklist
- [ ] _verify normalization works_
- [ ] _publish version_
- [ ] _update public docs and change log_
- [ ] _update developer logs_

### CODE REVIEW

**Review by airbyte-jenny — APPROVED** [review]
The decorator wrapping approach seems clever and pretty clean.  Code makes sense overall.

File: airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/SshWrappedJdbcDestination.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+/**
+ * Decorates a Destination with an SSH Tunnel using the standard configuration that Airbyte uses for
+ * configuring SSH.
+ */
+public class SshWrappedJdbcDestination implements Destination {
+
+  private final Destination delegate;
+
+  public SshWrappedJdbcDestination(final Destination delegate) {
```
Comment (michel-tricot) [inline airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/SshWrappedJdbcDestination.java:48]: According to the name of the class, if it is only for JDBC destination, shoudn't the argument be of type `JdbcDestination` or a less permissive type?

**Review by sherifnada — COMMENTED** [review]
I have the same question about transparently replacing hostname, but otherwise looks mergable

File: airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/SshWrappedJdbcDestination.java (hunk 1 above)
Comment (cgardens) [reply airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/SshWrappedJdbcDestination.java:48]: Good call. I actually think the name of this class should be more permissive. Just call it `SshWrappedDestination`. Ultimately there's nothing here going on that can't work for anything that we might want to access via ssh tunnel.

**Review by jrhizor — APPROVED** [review]

**Review by sherifnada — APPROVED** [review]
mostly reviewed the normalization changes, lgtm

**Pull request merged**